src/parser/gpx.rs

```diff
@@ -34,30 +58,13 @@ pub fn consume<R: Read>(context: &mut Context<R>) -> Result<Gpx, GpxError> {
     let mut description: Option<String> = None;
     let mut keywords: Option<String> = None;
 
-    // First we consume the gpx tag and its attributes
-    let attributes = verify_starting_tag(context, "gpx")?;
-    let version = attributes
-        .iter()
-        .find(|attr| attr.name.local_name == "version")
-        .ok_or(GpxError::InvalidElementLacksAttribute("version", "gpx"))?;
-    gpx.version = version_string_to_version(&version.value)?;
     context.version = gpx.version;
 
-    let creator = attributes
-        .iter()
-        .find(|attr| attr.name.local_name == "creator");
-    gpx.creator = creator.map(|c| c.value.clone());
-
     loop {
-        let next_event = {
-            if let Some(next) = context.reader.peek() {
-                match next {
-                    Ok(n) => n,
-                    Err(_) => return Err(GpxError::EventParsingError("Expecting an event")),
-                }
-            } else {
-                break;
-            }
+        let next_event = match context.reader.peek() {
```

Comment:
I like this (nested matching in `next_event`), but the `try_from_attributes` just looks like a not very successful attempt to shrink the function.

Not strongly against it, but it feels a bit pointless.